pinot-server/src/main/java/org/apache/pinot/server/api/resources/QueryResource.java

```diff
@@ -65,10 +67,20 @@ public class QueryResource {
       @ApiResponse(code = 404, message = "Query not found running on the server")
   })
   public String cancelQuery(
-      @ApiParam(value = "QueryId as in the format of <brokerId>_<requestId>", required = true) @PathParam("queryId")
-          String queryId) {
+      @ApiParam(value = "QueryId as in the format of <brokerId>_<requestId> or <brokerId>_<requestId>_(O|R)",
```

Comment:
This API is on the server side, which is meant to be called by the broker. We need the type suffix to separate the OFFLINE and REALTIME query on the server side. Since they are tracked as 2 separate queries, it is better to give finer control.
On the broker side, the query cancellation will take the request id and always cancel both queries. When the server is upgraded, broker can cancel OFFLINE queries and REALTIME queries individually because they might fanout to different servers.